components/src/core/Drawer/DrawerNavGroup.tsx

```diff
@@ -112,31 +243,153 @@ function NavigationListItem(item: NavItem, props: DrawerNavGroupProps): ReactNod
         }
     };
 
+    const chevron = itemChevron !== undefined ? itemChevron : groupChevron;
+
+    function getExpandIcon(): JSX.Element {
+        if (expandIcon) {
+            return expandIcon;
+        }
+        if (depth) {
+            return <ArrowDropUp />;
+        }
+        return <ExpandLess />;
+    }
+
+    function getActionComponent(): JSX.Element {
+        if (chevron) {
+            return null;
+        }
+        if (!subItems) {
+            return null;
+        }
+        if (collapseIcon) {
+            return (
+                <div
+                    onClick={(e): void => {
+                        if (e) {
+                            expandHandler();
+                        }
+                    }}
+                    className={classes.expandIcon}
+                >
+                    {expanded ? collapseIcon : getExpandIcon()}
+                </div>
+            );
+        }
+
+        return (
+            <div
+                onClick={(e): void => {
+                    if (e) {
+                        expandHandler();
+                    }
+                }}
+                className={`${classes.expandIcon} ${expanded ? 'expanded' : ''}`}
+            >
+                {getExpandIcon()}
+            </div>
+        );
```

Comment:
The value in these two returns blocks is very similar...can we combine them into a single return with a little conditional logic where needed?

Reply:
Yeah I thought about it. I wondered if I should trade that for readability.

Reply:
I think it's more readable to have a single return block